src/query_plan/fetch_dependency_graph_processor.rs

```diff
@@ -89,6 +87,7 @@ pub(crate) struct FetchDependencyGraphToCostProcessor;
 pub(crate) trait FetchDependencyGraphProcessor<TProcessed, TDeferred> {
     fn on_node(
         &mut self,
+        graph: &FetchDependencyGraph,
         node: &mut FetchDependencyGraphNode,
```

Comment:
Nodes are part of a FetchDependencyGraph, so this API requires a mutable reference and an immutable reference to be alive at the same time. we should provide a graph and a NodeIndex instead of a mutable reference to the node (this will cause a small conflict with #261).

Reply:
I’ve changed the parameter to only `&QueryGraph`. It should be possible to borrow disjoint fields of `FederatedDepnedencyGraph` at the same time.